src/main/java/com/epam/labtaskspringcore/utils/UsernameGeneratorImpl.java

```diff
@@ -25,42 +26,50 @@ public UsernameGeneratorImpl(TrainerDAO trainerDAO, TraineeDAO traineeDAO) {
     }
 
     public String generateUsername(User user) {
-
         String base = user.getFirstName() + "." + user.getLastName();
         int counter = 1;
-        StringBuilder username = new StringBuilder().append(base);
-        while (
-                !isUnique(user, username.toString(), trainerDAO.getTrainers()) ||
-                        !isUnique(user, username.toString(), traineeDAO.getTrainees())
-        ) {
-            username = new StringBuilder().append(base).append(counter);
-            counter++;
+        String username = base;
+        Function<String, Boolean> isUniqueInTrainers = u -> isUnique(user, u, trainerDAO.getTrainers());
+        Function<String, Boolean> isUniqueInTrainees = u -> isUnique(user, u, traineeDAO.getTrainees());
+
+        while (!isUniqueInTrainees.apply(username) || !isUniqueInTrainers.apply(username)) {
+            username = base + counter++;
         }
+
         log.info(">>>> Generating username");
-        return username.toString();
+        return username;
     }
 
-
-    private <T extends User, V extends User> boolean isUnique(V targetUser, String username, List<T> users) {
-        boolean result = true;
-        for (T user : users) {
-            if (user.getUsername().equals(username) && !user.equals(targetUser)) {
-                result = false;
-                break;
-            }
-        }
-        return result;
+    private <T extends User> boolean isUnique(User targetUser, String username, List<T> users) {
+        return users.stream().noneMatch(user -> user.getUsername().equals(username) && !user.equals(targetUser));
     }
 
-
-//    private <T extends User> boolean isUnique(String username, List<T> users) {
-//        boolean result = true;
-//        for (T user : users) {
-//            if (user.getUsername().equals(username)) {
-//                result = false;
-//                break;
-//            }
-//        }
-//        return result;
-//    }
+    //    public String generateUsername(User user) {
```

Comment:
Please delete code which you are not using

Reply:
improved with commit a26ad1d